northstar-tests/tests/console.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use api::{client::Error as ClientError, model::Error as ModelError};
+use futures::{SinkExt, StreamExt};
+use northstar::api::{
+    self,
+    model::{self, ConnectNack},
+};
+use northstar_tests::runtime_test;
+use tokio::{net::UnixStream, time::Duration};
+
+/// Connect a client to the runtime console without any permission configured.
+async fn connect_none() -> Result<api::client::Client<UnixStream>> {
+    let io = UnixStream::connect(&northstar_tests::runtime::console_none().path()).await?;
+    api::client::Client::new(io, None, Duration::from_secs(10))
+        .await
+        .context("Failed to connect to the runtime")
+}
+
+// Verify that the client() reject a version mismatch in Connect
+#[runtime_test]
+async fn api_version() -> Result<()> {
+    let mut connection = api::codec::Framed::new(
+        UnixStream::connect(&northstar_tests::runtime::console_none().path()).await?,
+    );
+
+    // Send a connect with an version unequal to the one defined in the model
+    let mut version = api::model::version();
+    version.patch += 1;
+
+    let connect = api::model::Connect::new_connect(version, false);
+    let connect_message = api::model::Message::new_connect(connect);
+    connection.send(connect_message.clone()).await?;
+
+    // Receive connect nack
+    let connack = connection.next().await.unwrap().unwrap();
+
+    drop(connection);
+
+    let error = ConnectNack::InvalidProtocolVersion {
+        version: model::version(),
+    };
+    let expected_message = model::Message::new_connect(model::Connect::Nack { error });
+
+    assert_eq!(connack, expected_message);
+    Ok(())
+}
+
+#[runtime_test]
+async fn notifications() -> Result<()> {
+    let io = UnixStream::connect(&northstar_tests::runtime::console_none().path()).await?;
+    assert!(
+        api::client::Client::new(io, Some(10), Duration::from_secs(10))
+            .await
+            .is_err()
```

Comment:
There's a comment missing. The test tries to connect with the `subscribe_notifications` flags set to to `true`. This is not allowed on the url with no permissions at all. The test cannot use the `client()` fn because that one connects to console with `full` permissions.

I will add a comment.

Reply:
Done